core/src/main/java/io/temporal/samples/hello/HelloUpdate.java

```diff
@@ -0,0 +1,281 @@
+       * mutated inside a query.
+       */
+      if (receivedMessages.size() >= 10) {
+        /*
+         * Throwing an exception inside an update validator will cause the update to be rejected.
+         * Note: rejected update will not appear in the workflow history
+         */
+        throw new IllegalStateException("Only 10 greetings may be added to the queue");
+      }
+    }
+
+    @Override
+    public void exit() {
+      exit = true;
+    }
+  }
+
+  /**
+   * With the Workflow and Activities defined, we can now start execution. The main method starts
+   * the worker and then the workflow.
+   */
+  public static void main(String[] args) throws Exception {
+
+    // Get a Workflow service stub.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
+
+    /*
+     * Get a Workflow service client which can be used to start, Signal, and Query Workflow Executions.
+     */
+    WorkflowClient client = WorkflowClient.newInstance(service);
+
+    /*
+     * Define the workflow factory. It is used to create workflow workers for a specific task queue.
+     */
+    WorkerFactory factory = WorkerFactory.newInstance(client);
+
+    /*
+     * Define the workflow worker. Workflow workers listen to a defined task queue and process
+     * workflows and activities.
+     */
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    /*
+     * Register the workflow implementation with the worker.
+     * Workflow implementations must be known to the worker at runtime in
+     * order to dispatch workflow tasks.
+     */
+    worker.registerWorkflowImplementationTypes(GreetingWorkflowImpl.class);
+
+    /*
+     * Register our Activity Types with the Worker. Since Activities are stateless and thread-safe,
+     * the Activity Type is a shared instance.
+     */
+    worker.registerActivitiesImplementations(new HelloActivity.GreetingActivitiesImpl());
+
+    /*
+     * Start all the workers registered for a specific task queue.
+     * The started workers then start polling for workflows and activities.
+     */
+    factory.start();
+
+    // Create the workflow options
+    WorkflowOptions workflowOptions =
+        WorkflowOptions.newBuilder().setTaskQueue(TASK_QUEUE).setWorkflowId(WORKFLOW_ID).build();
+
+    // Create the workflow client stub. It is used to start the workflow execution.
+    GreetingWorkflow workflow = client.newWorkflowStub(GreetingWorkflow.class, workflowOptions);
+
+    // Start workflow asynchronously and call its getGreeting workflow method
+    WorkflowClient.start(workflow::getGreetings);
+
+    // After start for getGreeting returns, the workflow is guaranteed to be started.
+    // So we can send an update to it using the workflow stub.
+    // This workflow keeps receiving updates until exit is called
+
+    // When the workflow is started the getGreetings will block for the previously defined
+    // conditions
+    // Send the first workflow update
+    workflow.waitForName("World");
```

Comment:
You demonstrate returning from an update but don't ever use the result

Reply:
The  result is used, just not by every call.